File: training-data/ruby/max_subarray.rb
def max_subarray(nums)
  best = current = nums.first
  nums[1..].each do |x|
    current = [x, current + x].max
    best = [best, current].max
  end
  best
end

puts max_subarray([-2, 1, -3, 4, -1, 2, 1, -5, 4])

File: training-data/python/max_subarray.py
def max_subarray(nums):
    best = current = nums[0]
    for x in nums[1:]:
        current = max(x, current + x)
        best = max(best, current)
    return best


if __name__ == "__main__":
    print(max_subarray([-2, 1, -3, 4, -1, 2, 1, -5, 4]))
